link/src/main/java/com/stripe/android/link/ui/wallet/WalletScreen.kt

```diff
@@ -0,0 +1,393 @@
+        ) {
+            itemsIndexed(paymentDetails) { index, item ->
+                when (item) {
+                    is ConsumerPaymentDetails.Card -> {
+                        CardPaymentMethodItem(
+                            cardDetails = item,
+                            isSelected = selectedIndex == index
+                        ) {
+                            onIndexSelected(index)
+                        }
+                    }
+                }
+            }
+        }
+        Row(
+            modifier = Modifier
+                .fillMaxWidth()
+                .height(60.dp)
+                .padding(start = horizontalPadding, end = horizontalPadding, bottom = 4.dp)
+                .clickable(onClick = onAddNewPaymentMethodClick),
+            verticalAlignment = Alignment.CenterVertically
+        ) {
+            Icon(
+                painter = painterResource(id = R.drawable.ic_link_add),
+                contentDescription = null,
+                modifier = Modifier.padding(end = 12.dp),
+                tint = Color.Unspecified
+            )
+            Text(
+                text = stringResource(id = R.string.wallet_add_new_payment_method),
+                color = MaterialTheme.colors.onPrimary,
+                style = MaterialTheme.typography.button
+            )
+        }
+    }
+}
+
+@Composable
+internal fun CardPaymentMethodItem(
+    cardDetails: ConsumerPaymentDetails.Card,
+    isSelected: Boolean,
+    onClick: () -> Unit
+) {
+    Row(
+        modifier = Modifier
+            .fillMaxWidth()
+            .height(56.dp)
+            .clickable(onClick = onClick),
+        verticalAlignment = Alignment.CenterVertically
+    ) {
+        RadioButton(
+            selected = isSelected,
+            onClick = null,
+            modifier = Modifier.padding(end = 6.dp),
+            colors = RadioButtonDefaults.colors(
+                selectedColor = MaterialTheme.colors.primary,
+                unselectedColor = MaterialTheme.colors.onBackground
+            )
+        )
+        CardDetails(card = cardDetails)
+        Spacer(modifier = Modifier.weight(1f))
+        if (cardDetails.isDefault) {
+            Box(
+                modifier = Modifier
+                    .height(20.dp)
+                    .background(
+                        color = MaterialTheme.colors.secondary,
+                        shape = MaterialTheme.shapes.small
+                    ),
+                contentAlignment = Alignment.Center
+            ) {
+                Text(
+                    text = stringResource(id = R.string.wallet_default),
+                    modifier = Modifier.padding(horizontal = 4.dp, vertical = 2.dp),
+                    color = MaterialTheme.colors.onBackground,
+                    fontSize = 12.sp,
+                    fontWeight = FontWeight.Medium
+                )
+            }
+        }
+    }
+    Divider(color = MaterialTheme.colors.onBackground, thickness = 1.dp)
```

Comment:
At some point we should make this divider consistent with other dividers (such as the one between the Google Pay and form.